asterius/rts/rts.float.mjs

```diff
@@ -0,0 +1,108 @@
+// Implements primitives from primFloat.c
+export class FloatCBits {
+    constructor(memory) {
+        this.memory  = memory;
+        /* Constants copy-pasted by running this C program:
+        #include <float.h>
+        #include <stdio.h>
+
+        #define PRINT(name) printf("this." #name " = %d;\n", name);
+        int main() {
+            PRINT(FLT_MIN_EXP); PRINT(FLT_MANT_DIG);
+            PRINT(DBL_MIN_EXP); PRINT(DBL_MANT_DIG);
+            return 0;
+        }
+        Other code copy-pasted from C calculations.
+        */
+
+
+        this.FLT_MIN_EXP = -125;
+        this.FLT_MANT_DIG = 24;
+        this.DBL_MIN_EXP = -1021;
+        this.DBL_MANT_DIG = 53;
+
+        this.MY_DMINEXP = ((this.DBL_MIN_EXP) - (this.DBL_MANT_DIG) - 1);
+        /* DMINEXP is defined in values.h on Linux (for example) */
+        this.DHIGHBIT = 0x00100000;
+        this.DMSBIT = 0x80000000;
+
+        this.MY_FMINEXP = ((this.FLT_MIN_EXP) - (this.FLT_MANT_DIG) - 1);
+        this.FHIGHBIT = 0x00800000;
+        this.FMSBIT = 0x80000000;
+        Object.seal(this);
+    }
+
+    // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/is
+    isFloatNegativeZero(x) {
+        return Object.is(-0, x);
+    }
+
+    isFloatNaN(x) {
+        return x != x;
+    }
+    isFloatInfinite(x) {
+        return !isFinite(x);
+    }
+
+    // Does it really make sense to have two functions?  probably not...
+    isDoubleNegativeZero(x) {
+        return Object.is(-0, x);
+    }
+
+    // https://stackoverflow.com/questions/2003493/javascript-float-from-to-bits
+    FloatToIEEE(f) {
```

Comment:
How about using `DataView` instead to implement reinterpret-cast from `f32` to `i32`? The code will be cleaner compared to different typed arrays using the same buffer.

Also, since `FloatToIEEE` won't be called in multiple threads, it would be nice to promote the local buffer to a class member to reduce the allocation overhead a bit.